// jw-angular-social-buttons/lib/tweet.component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var TweetComponent = /** @class */ (function () {
    function TweetComponent() {
        this.url = location.href;
        this.text = '';
        // load twitter sdk if required
        var url = 'https://platform.twitter.com/widgets.js';
        if (!document.querySelector("script[src='" + url + "']")) {
            var script = document.createElement('script');
            script.src = url;
            document.body.appendChild(script);
        }
    }
    TweetComponent.prototype.ngAfterViewInit = function () {
        // render tweet button
        window['twttr'] && window['twttr'].widgets.load();
    };
    TweetComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'tweet',
                    template: "<a href=\"https://twitter.com/share\" [attr.data-text]=\"text\" [attr.data-url]=\"url\" class=\"twitter-share-button\"></a>"
                },] },
    ];
    /** @nocollapse */
    TweetComponent.ctorParameters = function () { return []; };
    TweetComponent.propDecorators = {
        "url": [{ type: core_1.Input },],
        "text": [{ type: core_1.Input },],
    };
    return TweetComponent;
}());
exports.TweetComponent = TweetComponent;
